### PR TITLE
fix: Don't allow multipleInstance to be set on homework

### LINKED ISSUE #5646: Cause error on sync for homeworks with multipleInstance set to true
Currently homeworks to not support multiple instance:

https://github.com/PrairieLearn/PrairieLearn/blob/dfa65a05d9f00563d02dc6146f03de95b0ea591f/pages/studentAssessmentHomework/studentAssessmentHomework.js#L19-L25

However this isn't warned in sync. This kind of warning should be pretty straightforward to add and could avoid confusion in runtime. So I suggest that sync give an error message if exam type is Homework and `multipleInstance` is true.

### PR DESCRIPTION
**Pull request by tanyongzhi**
Fixes #5646 

Check if homework has multiple instances set to true during sync. If so, throw an error message during the sync itself, instead of when a student accesses the homework.

Example of a "Homework" assessment with multiple instances set to true:
image

### CODE REVIEW

**Comment by tanyongzhi** [comment]
_**I have reviewed and hereby sign the CLA**_

**Comment by tanyongzhi** [comment]
recheck

**Comment by tanyongzhi** [comment]
**recheck**

**Comment by nwalters512** [comment]
@tanyongzhi thanks for the contribution! You'll need to copy the CLA statement without the bold/italic formatting. I understand this is pretty confusing - we're still waiting for https://github.com/contributor-assistant/github-action/pull/104 to be merged!

**Comment by tanyongzhi** [comment]
I have reviewed and hereby sign the CLA

**Comment by tanyongzhi** [comment]
recheck

File: sync/course-db.js [hunk 1]
```diff
@@ -1105,8 +1105,13 @@ async function validateAssessment(assessment, questions) {
...
   // real-time grading to be disabled for them.
   const allowRealTimeGrading = _.get(assessment, 'allowRealTimeGrading', true);
-  if (!allowRealTimeGrading && assessment.type === 'Homework') {
-    errors.push(`Real-time grading cannot be disabled for Homework-type assessments`);
+  if (assessment.type === 'Homework') {
+    if (!allowRealTimeGrading) {
+      errors.push(`Real-time grading cannot be disabled for Homework-type assessments`);
+    }
+    if (assessment.multipleInstance) {
+      errors.push(`Homework cannot have multiple instances`);
```
Comment (nwalters512) [inline sync/course-db.js:1113]: This makes the error message a little more consistent with the others and hopefully speeds up resolution be referring to the specific option:
```suggestion
      errors.push(`"multipleInstance" cannot be used for Homework-type assessments`);
```

**Review by nwalters512 — COMMENTED** [review]
Please add a test for this behavior. If you want to pattern match against something, there are a lot of existing tests that you can look at, e.g. 

https://github.com/PrairieLearn/PrairieLearn/blob/d240fcf137acb56464fe29d1fcb11949ed0e7dc7/tests/sync/assessmentsSync.test.js#L681-L695

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**